swap_meet/vendor.py

```diff
@@ -1,2 +1,218 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory else []
+
+    def add(self, item):
+        """
+        Adds item to Vendor.inventory.
+        :params:
+        - (obj) item: item to add
+        :returns:
+        - (obj) item added
+        """
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        """
+        Remove item to Vendor.inventory.
+        :params:
+        - (obj) item: to remove
+        :returns:
+        - (obj) item removed or None
+        """
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return None
+
+    def get_by_id(self, id):
+        """
+        Retrieves item by id.
+        :params:
+        - (int) id: id of item
+        :returns:
+        - (obj) item with id or None
+
+        """
+        result = list(filter(lambda item: item.id == id, self.inventory))
+        return result[0] if result else None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        """
+        Swaps two indicated items between vendors.
+        :params:
+        - (obj) other_vendor: other vendor to switch with
+        - (obj) my_item: my item to switch
+        - (obj) their_item: other vendor's item to switch
+        :returns:
+        - (bool) True if items swapped or False
+        """
+        if (my_item not in self.inventory or
+           their_item not in other_vendor.inventory):
+            return False
+        else:
+            self.remove(my_item)
```

Comment:
well done making use of existing vendor functions, rather than reimplementing the logic here